examples/finite_prime_field.rs

```diff
@@ -0,0 +1,256 @@
+use noether::{
+    AssociativeAddition, AssociativeMultiplication, CommutativeAddition, CommutativeMultiplication,
+    DistributiveAddition, DistributiveMultiplication, FiniteField,
+};
+use num_traits::{Euclid, Inv, One, Zero};
+use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, Sub, SubAssign};
+
+/// The trait hierarchy in Noether is useful for verifying the correct and idiomatic implementation
+/// of algebraic structures in rust.
+///
+/// We take here, as an example, the implementation of a finite field.
+
+#[derive(Clone, Copy, Debug)]
+pub struct FinitePrimeField<const L: usize, const D: usize> {
+    modulus: [u64; L],
+    _value: [u64; L],
+}
+
+impl<const L: usize, const D: usize> FinitePrimeField<L, D> {
+    const _ZERO: [u64; L] = Self::zero_array();
+
+    pub const fn new(modulus: [u64; L], value: [u64; L]) -> Self {
+        if D != 2 * L {
+            panic!("Double size D must be twice the size of the field L");
+        }
+        Self {
```

Comment:
If L is the order of the field, I suppose we could also verify that it's of the form p^n.

Reply:
The modulus is the order of the field, and yes.